src/helics/apps/Connector.cpp

```diff
@@ -287,6 +292,63 @@ nlohmann::json TemplateMatcher::usedInterfaceGeneration()
     return generator;
 }
 
+static bool addPotentialTargets(ConnectionsList& connections,
+                                const std::string& interfaceName,
+                                std::string targetFieldName,
+                                const nlohmann::json& interface)
+{
+    bool found{false};
+    // There should probably be a static_assert here but there isn't a nice type trait to check that
+    if (interface.contains(targetFieldName)) {
+        auto targets = interface[targetFieldName];
+        if (targets.is_array()) {
+            for (const auto& target : targets) {
+                connections.potentialInterfaceConnections.emplace_back(interfaceName,
+                                                                       target.get<std::string>());
+            }
+        } else {
+            connections.potentialInterfaceConnections.emplace_back(interfaceName,
+                                                                   targets.get<std::string>());
+        }
+        found = true;
+    }
+    if (targetFieldName.back() == 's') {
+        targetFieldName.pop_back();
+        if (interface.contains(targetFieldName)) {
+            connections.potentialInterfaceConnections.emplace_back(
+                interfaceName, interface[targetFieldName].get<std::string>());
+            found = true;
+        }
+    }
+    return found;
+}
+
+static bool addPotentialTargetVariations(ConnectionsList& connections,
```

Comment:
The "names" for targets that are equivalent feel like they should be documented, unless colliding potential target names isn't a concern.

Reply:
The connector is mainly concerned about making connections, so they all end up equivalent even if there is slightly different meanings of the different variations in the actual constructor.  Even in that case duplication is fine and in some cases intentional.   But we probably should check on the documentation of that topic to make sure it is clear.